src/base/bittorrent/session.cpp

```diff
@@ -408,9 +408,9 @@ Session::Session(QObject *parent)
     , m_stopTrackerTimeout(BITTORRENT_SESSION_KEY("StopTrackerTimeout"), 1)
     , m_isSuperSeedingEnabled(BITTORRENT_SESSION_KEY("SuperSeedingEnabled"), false)
     , m_maxConnections(BITTORRENT_SESSION_KEY("MaxConnections"), 500, lowerLimited(0, -1))
-    , m_maxUploads(BITTORRENT_SESSION_KEY("MaxUploads"), -1, lowerLimited(0, -1))
+    , m_maxUploads(BITTORRENT_SESSION_KEY("MaxUploads"), 20, lowerLimited(0, -1))
     , m_maxConnectionsPerTorrent(BITTORRENT_SESSION_KEY("MaxConnectionsPerTorrent"), 100, lowerLimited(0, -1))
-    , m_maxUploadsPerTorrent(BITTORRENT_SESSION_KEY("MaxUploadsPerTorrent"), -1, lowerLimited(0, -1))
+    , m_maxUploadsPerTorrent(BITTORRENT_SESSION_KEY("MaxUploadsPerTorrent"), 4, lowerLimited(0, -1))
```

Comment:
@an0n666 
I suppose https://github.com/arvidn/libtorrent/pull/4417 won't be finished in a short time. I'm thinking of separating this change (MaxUploadsPerTorrent) and the one above (MaxUploads) to another PR so we can have it in qbt sooner. The changes below (ChokingAlgorithm) will stay in this PR.
What do you think?

Reply:
Okay.

Reply:
See PR #12526.